.pre-commit-config.yaml

```diff
@@ -0,0 +1,23 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+- repo: https://github.com/astral-sh/ruff-pre-commit
+  rev: v0.1.14
+  hooks:
+    # Run the linter.
+  - id: ruff
+    # Run the formatter.
+  - id: ruff-format
+    args: [--check]
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.5.0
+  hooks:
+  - id: trailing-whitespace
+  - id: end-of-file-fixer
+  - id: check-yaml
+  - id: check-toml
+  - id: check-ast
+  - id: check-added-large-files
+  - id: check-merge-conflict
+  - id: check-shebang-scripts-are-executable
```

Comment:
As I removed the `setup.py` file (GitHub can now read the name in `pyproject.toml` :tada:), there are no executables in the repository. We can drop these two checks.

Reply:
sure.